### ISSUE TITLE
/ccalc bug

### ISSUE THREAD

**Issue by Alexresh** [Earthcomputer/clientcommands#198]
1.16.5 Lasted commit
![image](https://user-images.githubusercontent.com/9035526/110209977-78b61c80-7ea0-11eb-84ed-909cf5123634.png)

**Comment by Earthcomputer** [comment]
Hmm, interesting

**Comment by Vonr** [comment]
maybe it was interpreted as -1-(1-1) or -(1-1)-1

**Comment by Alexresh** [comment]
![image](https://user-images.githubusercontent.com/9035526/110232161-32f36580-7f2d-11eb-86bd-c8e21340b8f5.png)

**Comment by Vonr** [comment]
try calculating (0-1)-1 and (-1-1)-1

**Comment by Spacebarium** [comment]
![image](https://user-images.githubusercontent.com/71394206/111421641-4d48f280-8728-11eb-94d5-19cef8705499.png)

![image](https://user-images.githubusercontent.com/71394206/111421921-c5171d00-8728-11eb-93fc-d5c8482337fd.png)

interesting patterns

**Comment by James103** [comment]
Similarly, we have the following results with repeated division:
`/ccalc 2/2` = 1 as expected.
`/ccalc 2/2/2` = 2 since 2/(2/2) = 2/1 = 2. Expected value is (2/2)/2 = 1/2 = 0.5
`/ccalc 2/2/2/2` = 1 since 2/(2/(2/2)) = 2/2 = 1. Expected value is ((2/2)/2)/2 = 1/4 = 0.25
`/ccalc 2/2/2/2/2` = 2 since 2/(2/(2/(2/2))) = 2/1 = 2. Expected value is (((2/2)/2)/2)/2 = 1/8 = 0.125
`/ccalc 2/2/2/2/2/2` = 1 since 2/(2/(2/(2/(2/2)))) = 2/2 = 1. Expected value is ((((2/2)/2)/2)/2)/2 = 1/16 = 0.0625
...

In the case of subtraction:
`/ccalc -1-1` = -2 as expected, though it should be parsed as (-1)-1 = -2.
`/ccalc -1-1-1` = -1 since -1-(1-1) = -1. Expected value is ((-1)-1)-1 = -2-1 = -3.
`/ccalc -1-1-1-1` = -2 since -1-(1-(1-1)) = -2. Expected value is (((-1)-1)-1)-1 = -4.
`/ccalc -1-1-1-1-1` = -1 since -1-(1-(1-(1-1))) = -1. Expected value is ((((-1)-1)-1)-1)-1 = -5.
`/ccalc -1-1-1-1-1-1` = -2 since -1-(1-(1-(1-(1-1)))) = -2. Expected value is (((((-1)-1)-1)-1)-1)-1 = -6.
...

In other words, `/ccalc` evaluates expressions from right to left, instead of the left to right expected from the order of operations.

**Comment by xpple** [comment]
> Similarly, we have the following results with repeated division:
> `/ccalc 2/2` = 1 as expected.
> `/ccalc 2/2/2` = 2 since 2/(2/2) = 2/1 = 2. Expected value is (2/2)/2 = 1/2 = 0.5
> `/ccalc 2/2/2/2` = 1 since 2/(2/(2/2)) = 2/2 = 1. Expected value is ((2/2)/2)/2 = 1/4 = 0.25
> `/ccalc 2/2/2/2/2` = 2 since 2/(2/(2/(2/2))) = 2/1 = 2. Expected value is (((2/2)/2)/2)/2 = 1/8 = 0.125
> `/ccalc 2/2/2/2/2/2` = 1 since 2/(2/(2/(2/(2/2)))) = 2/2 = 1. Expected value is ((((2/2)/2)/2)/2)/2 = 1/16 = 0.0625
> ...
> 
> In the case of subtraction:
> `/ccalc -1-1` = -2 as expected, though it should be parsed as (-1)-1 = -2.
> `/ccalc -1-1-1` = -1 since -1-(1-1) = -1. Expected value is ((-1)-1)-1 = -2-1 = -3.
> `/ccalc -1-1-1-1` = -2 since -1-(1-(1-1)) = -2. Expected value is (((-1)-1)-1)-1 = -4.
> `/ccalc -1-1-1-1-1` = -1 since -1-(1-(1-(1-1))) = -1. Expected value is ((((-1)-1)-1)-1)-1 = -5.
> `/ccalc -1-1-1-1-1-1` = -2 since -1-(1-(1-(1-(1-1)))) = -2. Expected value is (((((-1)-1)-1)-1)-1)-1 = -6.
> ...
> 
> In other words, `/ccalc` evaluates expressions from right to left, instead of the left to right expected from the order of operations.

I do not agree to this. Writing `2/2/2` or any similar series of division without brackets is ambiguous, and should thus not be relied on. However, the issues with any series of subtraction without brackets is clearly a bug, and should indeed be fixed.

**Comment by Earthcomputer** [comment]
Not sure why but I find this issue really funny :laughing:

**Comment by xpple** [comment]
> Not sure why but I find this issue really funny 😆

Because realistically no one uses the command either way? 🤣

**Comment by Earthcomputer** [comment]
I use it all the time, super convenient to have a calculator without switching away from MC

**Comment by xpple** [comment]
> I use it all the time, super convenient to have a calculator without switching away from MC

Yeah right, in case you forgot what `-1-1-1-1-1` is 😂

**Comment by Vonr** [comment]
I can't give a proper solution but one bad one would be to include a library that solves expressions for you

**Comment by Vonr** [comment]
it would be stupidly funny if this works but maybe wrap every constant in parentheses before solving them?
that way -1-1-1 would be solved as (-(1)-(1))-(1) which i cant see going wrong

**Comment by haykam821** [comment]
This is how the expression is parsed:

![Parsed expression: ((negated: (literal: 1.0)) - ((literal: 1.0) - (literal: 1.0)))](https://user-images.githubusercontent.com/24855774/111894544-ebb3ad00-89e1-11eb-806a-46cf38b1bdac.png)

Basically, it is parsed as `-1 - (1 - 1)`.

**Comment by James103** [comment]
@haykam821 How exactly did you get the "Parsed expression" message?

**Comment by haykam821** [comment]
@James103 I ran the command `/ccalc --parse -1-1-1` using the `--parse` flag that #215 adds.
